Sources/JWTKit/Vendor/AppleIdentityToken.swift

```diff
@@ -1,6 +1,17 @@
 /// - See Also:
 /// [Retrieve the User’s Information from Apple ID Servers](https://developer.apple.com/documentation/signinwithapplerestapi/authenticating_users_with_sign_in_with_apple)
 public struct AppleIdentityToken: JWTPayload {
+
+    /// Taken from https://developer.apple.com/documentation/authenticationservices/asuserdetectionstatus
+    public enum UserDetectionStatus : Int, Codable {
```

Comment:
I'm a little hesitant to introduce a new public enum here, especially to model something we don't have control over. The reason being that if Apple release a new status type, that's technically a break change for us if we add the new case to the enum. 

The best way around this would be to keep the public enum, but just make each case a `static let` property so adding new properties is non-breaking. We can then use an internal enum for decoding and storage, as adding new cases to that is non-breaking and then map between the enum cases and static properties. Does that makes sense?

Reply:
Conceptually I got it, but I haven't done it before that way, so is this something that you're asking @0xTim?

```
public enum UserDetectionStatus: Int, Codable {
    // Static let properties for each case
    public static let unsupported = UserDetectionStatus(rawValue: 0)
    public static let unknown = UserDetectionStatus(rawValue: 1)
    public static let likelyReal = UserDetectionStatus(rawValue: 2)

    // Internal enum for decoding and storage
    internal enum RawValue: Int, Codable {
        case unsupported
        case unknown
        case likelyReal
    }

    // Mapping between the enum cases and static properties
    public init(from decoder: Decoder) throws {
        let rawValue = try decoder.singleValueContainer().decode(RawValue.self)
        switch rawValue {
        case .unsupported:
            self = UserDetectionStatus.unsupported
        case .unknown:
            self = UserDetectionStatus.unknown
        case .likelyReal:
            self = UserDetectionStatus.likelyReal
        }
    }

    public func encode(to encoder: Encoder) throws {
        var container = encoder.singleValueContainer()
        switch self {
        case .unsupported:
            try container.encode(RawValue.unsupported)
        case .unknown:
            try container.encode(RawValue.unknown)
        case .likelyReal:
            try container.encode(RawValue.likelyReal)
        }
    }
}
```

Reply:
@qalandarov yep exactly!

Reply:
Had to make slight updates compared to the above example